### PR TITLE
#438 Add model for static var compensator

### PR DESCRIPTION
**Pull request by MarianneSaugierRTE**

### CODE REVIEW

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/CalculBG.mo [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+  ModeConnector mode "Current mode of the static var compensator" annotation(
+    Placement(visible = true, transformation(origin = {62, 120}, extent = {{-20, -20}, {20, 20}}, rotation = -90), iconTransformation(origin = {72, 110}, extent = {{-10, -10}, {10, 10}}, rotation = -90)));
+
+equation
+
+  BPu = if mode.value == Mode.RUNNING_V then BVarPu + BShuntPu elseif mode.value == Mode.STANDBY then BShuntPu else 0;
+
+  GPu = if mode.value == Mode.OFF then 0 else GCstPu;
+
```
Comment (adriguir) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/CalculBG.mo]: Open question again: basically you have two ways of writing these two if-loops. Either separately as you have done or in a unique loop such as:
if mode.value == Mode.RUNNING_V then
 BPu = BVarPu + BShuntPu;
 GPu = 0;
elseif 
 ....

In the second case, you see the overall SVarC behavior in each mode at first glance but if you prefer the current implementation, you can stick to it.

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/Limitations.mo [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+  Blocks.Math.Min min annotation(
+    Placement(visible = true, transformation(origin = {62, 40}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Blocks.Math.Max max annotation(
+    Placement(visible = true, transformation(origin = {62, -40}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Blocks.Sources.Constant bMaxPu(k = BMaxPu) annotation(
+    Placement(visible = true, transformation(origin = {16, 70}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Blocks.Sources.Constant bMinPu(k = BMinPu) annotation(
+    Placement(visible = true, transformation(origin = {18, -70}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Blocks.Continuous.LimIntegrator limIntegrator1(k = KCurrentLimiter, outMax = BMaxPu, outMin = 0, y_start = BMaxPu)  annotation(
```
Comment (adriguir) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/Limitations.mo]: limIntegratorMax?

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/Limitations.mo [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    Placement(visible = true, transformation(origin = {18, 34}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Blocks.Continuous.LimIntegrator limIntegrator2(k = KCurrentLimiter, outMax = 0, outMin = BMinPu, y_start = BMinPu)  annotation(
```
Comment (adriguir) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/Limitations.mo]: limIntegratorMin?

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/ModeHandling.mo [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    Placement(visible = true, transformation(origin = {110, -60}, extent = {{-10, -10}, {10, 10}}, rotation = 0), iconTransformation(origin = {110, -60}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Interfaces.BooleanOutput blocked "Wheter the static var compensator is blocked due to very low voltages" annotation(
+    Placement(visible = true, transformation(origin = {110, 60}, extent = {{-10, -10}, {10, 10}}, rotation = 0), iconTransformation(origin = {110, 60}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+
+  ModeConnector mode(value(start = Mode0)) "Current mode of the static var compensator" annotation(
+    Placement(visible = true, transformation(origin = {100, 0}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {110, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  ModeConnector mode_auto(value(start = Mode0)) "Mode of the static var compensator when in automatic configuration";
+  ModeConnector mode_manual(value(start = Mode0)) "Mode of the static var compensator when in manual configuration";
```
Comment (adriguir) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/ModeHandling.mo]: Coding style: modeAutomatic or automaticMode? modeManual or manualMode?

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/ModeHandling.mo [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+  Modelica.Blocks.Interfaces.IntegerInput setMode "Mode selected when in manual configuration" annotation(
```
Comment (adriguir) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/ModeHandling.mo]: Add manual somewhere in the variable name for clarity purpose

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/ModeHandling.mo [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+    Placement(visible = true, transformation(origin = {-120, -20}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {-120, 80}, extent = {{-20, -20}, {20, 20}}, rotation = 0)));
+  Modelica.Blocks.Interfaces.BooleanInput selectModeAuto "Wheter the static var compensator is in automatic configuration" annotation(
+    Placement(visible = true, transformation(origin = {-120, -60}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {-120, 28}, extent = {{-20, -20}, {20, 20}}, rotation = 0)));
+
+  Types.Time timerModeChangeUp(start = Modelica.Constants.inf) "Timer for the transition from standby to running mode for high voltage values";
+  Types.Time timerModeChangeDown(start = Modelica.Constants.inf) "Timer for the transition from standby to running mode for low voltage values";
+  Types.VoltageModule URef_auto(start = URef0) "Voltage reference for the regulation if the static var compensator switches from standy to running mode automatically in kV";
```
Comment (adriguir) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/ModeHandling.mo]: Coding style

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/ModeHandling.mo [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+  parameter Types.VoltageModule URef0  "Start value of voltage reference in kV";
+  parameter Mode Mode0 "Start value for mode";
+
+equation
+  // Timer for the transition from standby to running mode for high voltage values
+  when UPu*UNom > UThresholdUp then
+    timerModeChangeUp = time;
+  elsewhen UPu*UNom < UThresholdUp then
+    timerModeChangeUp = Modelica.Constants.inf;
+  end when;
```
Comment (adriguir) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/ModeHandling.mo]: We can easily avoid a few operations there without loosing in clarity I think by dividing the threshold by UNom (doing a final parameter) and always compare in p.u.
-> it avoids all the multiplication (the optimisation is perhaps done by OM but I'm not sure: too be double checked with @rosiereflo )

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/ModeHandling.mo [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+  end if;
+  // Manual mode setting
+  if setMode == 1 then
+    mode_manual.value = Mode.OFF;
+  elseif setMode == 2 then
+    mode_manual.value = Mode.STANDBY;
+  elseif setMode == 3 then
+    mode_manual.value = Mode.RUNNING_V;
+  else
+    assert(false, "Failed to convert setMode value into mode (enum: 1:OFF, 2:STANDBY, 3:RUNNING_V");
```
Comment (adriguir) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/ModeHandling.mo]: One ) missing after RUNNING_V

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/Parameters.mo [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+* This file is part of Dynawo, an hybrid C++/Modelica open source time domain simulation tool for power systems.
+*/
+
+package Parameters "Parameters of the static var compensator"
+
+  record Params_Regulation
+    import Dynawo.Types;
+    parameter Types.ApparentPowerModule SNom "Static Var Compensator nominal apparent power in MVA";
+    parameter Types.PerUnit Lambda "Statism of the regulation law URefPu = UPu + Lambda*QPu in p.u (base UNom, SNom)";
+    parameter Types.PerUnit KG "Global gain of the regulation";
```
Comment (adriguir) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/Parameters.mo]: Kg?

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/Parameters.mo [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+    import Dynawo.Types;
+    parameter Types.PerUnit BShuntPu "Fixed susceptance of the static var compensator in p.u (for standby mode) (base SNom)";
+  end Params_CalculBG;
+
+  record Params_ModeHandling
+    import Dynawo.Types;
+    parameter Types.VoltageModule UNom "Static var compensator nominal voltage in kV";
+    parameter Types.VoltageModule UBlock "Voltage value under which the static var compensator is blocked";
+    parameter Types.VoltageModule UDeblockUp "Upper voltage value defining the voltage interval in which the static var compensator is deblocked";
+    parameter Types.VoltageModule UDeblockDown "Lower voltage value defining the voltage interval in which the static var compensator is deblocked";
```
Comment (adriguir) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/Parameters.mo]: I'm not very sure of Deblock. Certainly unblock but it leads to an uggly name UUnblockUp

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/SVarCControl.mo [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+  Modelica.Blocks.Interfaces.RealInput UPu(start = U0Pu) "Voltage at the static var compensator terminal in p.u (base UNom)" annotation(
+    Placement(visible = true, transformation(origin = {-168, -2}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {-115, -41}, extent = {{-15, -15}, {15, 15}}, rotation = 0)));
+  Modelica.Blocks.Interfaces.RealInput QInjPu(start = -Q0Pu) "Reactive power injected by the static var compensator in p.u (base SnRef) (generator convention)" annotation(
+    Placement(visible = true, transformation(origin = {-168, -68}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {-115, -83}, extent = {{-15, -15}, {15, 15}}, rotation = 0)));
+
+  Modelica.Blocks.Interfaces.BooleanInput selectModeAuto(start = true) "Mode selected when in manual configuration" annotation(
+    Placement(visible = true, transformation(origin = {-168, 54}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {-115, 49}, extent = {{-15, -15}, {15, 15}}, rotation = 0)));
+  Modelica.Blocks.Interfaces.IntegerInput setMode(start = 2) "Wheter the static var compensator is in automatic configuration" annotation(
```
Comment (adriguir) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/SVarCControl.mo]: Is it normal to have these default start values?

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/SVarCControl.mo [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+    Placement(visible = true, transformation(origin = {-168, 84}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {-115, 9}, extent = {{-15, -15}, {15, 15}}, rotation = 0)));
+
+  Modelica.Blocks.Interfaces.RealOutput BPu(start = B0Pu) "Susceptance of the static var compensator in p.u (base SNom)" annotation(
+    Placement(visible = true, transformation(origin = {146, -8}, extent = {{-10, -10}, {10, 10}}, rotation = 0), iconTransformation(origin = {110, 50}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Interfaces.RealOutput GPu(start = G0Pu) "Conductance of the static var compensator in p.u (base SNom)" annotation(
+    Placement(visible = true, transformation(origin = {146, -28}, extent = {{-10, -10}, {10, 10}}, rotation = 0), iconTransformation(origin = {110, -50}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+
+  Modelica.Blocks.Sources.Constant GPu_cst(k = G0Pu)  annotation(
```
Comment (adriguir) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/SVarCControl.mo]: Coding style

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/SVarCControl.mo [hunk 13]
```diff
@@ -0,0 +1,106 @@
...
+    Placement(visible = true, transformation(origin = {32, -42}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Regulation regulation(BMaxPu = BMaxPu, BMinPu = BMinPu,BVar0Pu = BVar0Pu, IMaxPu = IMaxPu, IMinPu = IMinPu, KCurrentLimiter = KCurrentLimiter, KG = KG, Kp = Kp, Lambda = Lambda, SNom = SNom, Ti = Ti)  annotation(
```
Comment (adriguir) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/SVarCControl.mo]: extra space missing BMinPu, BVar0Pu to increase readability

File: dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Logs/package.order [hunk 14]
```diff
@@ -1,2 +1,6 @@
 Timeline
 Constraint
+ConstraintKeys
+LogKeys
+TimelineKeys
+ErrorKeys
```
Comment (adriguir) [inline dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Logs/package.order]: Voluntarily commited?

File: nrt/data/SVarC/description/SVarC.tex [hunk 15]
```diff
@@ -0,0 +1,410 @@
...
+\subsubsection{Control model}
+
+The control model is all per-united in the SNom base.\\
+
+The conductance represents the active power losses of the static var compensator. Here we neglect the losses so $G_{Pu} = 0$.\\
+
+The susceptance represents the reactive power contribution of the static var compensator. The susceptance computation in per-unit depends on the static var compensator regulation mode:
+
+\begin{itemize}
+\item if mode = OFF, $B_{Pu} = 0$. The static var compensator doesn't injects reactive power;
```
Comment (adriguir) [inline nrt/data/SVarC/description/SVarC.tex]: typo -> inject

File: nrt/data/SVarC/description/SVarC.tex [hunk 16]
```diff
@@ -0,0 +1,410 @@
...
+
+The variable susceptance $BVar_{Pu}$ is limited thanks to limitations taking into account static susceptance limits and variable current limits.
+
+\subsection{Scenarios}
+The simulated scenarios are :
+\begin{itemize}
+\item a step on the reference voltage;
+\item a small reactive load variation;
+\item a large reactive load variation;
+\item a manuel mode change;
```
Comment (adriguir) [inline nrt/data/SVarC/description/SVarC.tex]: typo -> manual
typo -> impedant

File: nrt/data/SVarC/description/SVarC.tex [hunk 17]
```diff
@@ -0,0 +1,410 @@
...
+
+From $t=3s$ to $t=4s$, the SVC is switched off: its mode is "OFF", its susceptance is zero and no reactive power is injected to the grid.\\
+
+From $t=4s$ on, the SVC is back in running mode and starts regulating the voltage again.
+
+
+\newpage
+\subsection{Impedent three-phase fault}
+
+The SVC is initially running. An impedent 100ms three-phase short circuit is simulated from $t=1s$ to $t=1,1s$.
```
Comment (adriguir) [inline nrt/data/SVarC/description/SVarC.tex]: typo -> impedant

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/CMakeLists.txt [hunk 18]
```diff
@@ -0,0 +1,32 @@
...
+
+
+SET(MODEL_FILES
+    # Voltage control models
+    package.mo
+    SVarCControl.mo
+    SVarCControl.extvar
+    SVarCControl_INIT.mo
+    Parameters.mo
+    CalculBG.mo
```
[the remark this answers is not in the document]
Comment (MarianneSaugierRTE) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/CMakeLists.txt]: too long, i stick to CalculBG (abbreviation)

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/CalculBG.mo [hunk 19]
```diff
@@ -0,0 +1,43 @@
...
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, you can obtain one at http://mozilla.org/MPL/2.0/.
+* SPDX-License-Identifier: MPL-2.0
+*
+* This file is part of Dynawo, an hybrid C++/Modelica open source time domain simulation tool for power systems.
+*/
+
+model CalculBG "Final calculation of the susceptance and the conductance"
+  import Modelica;
+  import Dynawo.Electrical.Controls.Voltage.StaticVarCompensator.Parameters;
```
[the remark this answers is not in the document]
Comment (MarianneSaugierRTE) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/CalculBG.mo]: no, fixed

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/CalculBG.mo (hunk 1 above)
Comment (MarianneSaugierRTE) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/CalculBG.mo]: good idea, fixed

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/Limitations.mo (hunk 2 above)
Comment (MarianneSaugierRTE) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/Limitations.mo]: fixed

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/Limitations.mo (hunk 3 above)
Comment (MarianneSaugierRTE) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/Limitations.mo]: fixed

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/ModeHandling.mo (hunk 4 above)
Comment (MarianneSaugierRTE) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/ModeHandling.mo]: fixed: modeAuto and modeManual (I want it to start with mode)

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/ModeHandling.mo (hunk 5 above)
Comment (MarianneSaugierRTE) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/ModeHandling.mo]: fixed, replaced by setModeManual

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/ModeHandling.mo (hunk 6 above)
Comment (MarianneSaugierRTE) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/ModeHandling.mo]: fixed, URefAuto

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/ModeHandling.mo (hunk 7 above)
Comment (MarianneSaugierRTE) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/ModeHandling.mo]: fixed

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/ModeHandling.mo (hunk 8 above)
Comment (MarianneSaugierRTE) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/ModeHandling.mo]: fixed

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/Parameters.mo (hunk 9 above)
Comment (MarianneSaugierRTE) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/Parameters.mo]: fixed

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/Parameters.mo (hunk 10 above)
Comment (MarianneSaugierRTE) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/Parameters.mo]: fixed

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/SVarCControl.mo (hunk 12 above)
Comment (MarianneSaugierRTE) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/SVarCControl.mo]: fixed: GPuCst

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/SVarCControl.mo (hunk 13 above)
Comment (MarianneSaugierRTE) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/SVarCControl.mo]: fixed

File: nrt/data/SVarC/description/SVarC.tex (hunk 15 above)
Comment (MarianneSaugierRTE) [reply nrt/data/SVarC/description/SVarC.tex]: fixed

File: nrt/data/SVarC/description/SVarC.tex (hunk 16 above)
Comment (MarianneSaugierRTE) [reply nrt/data/SVarC/description/SVarC.tex]: fixed

File: nrt/data/SVarC/description/SVarC.tex (hunk 17 above)
Comment (MarianneSaugierRTE) [reply nrt/data/SVarC/description/SVarC.tex]: fixed

File: dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Logs/package.order (hunk 14 above)
Comment (MarianneSaugierRTE) [reply dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Logs/package.order]: no... automatic modification from OM...

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/SVarCControl.mo (hunk 11 above)
Comment (MarianneSaugierRTE) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/StaticVarCompensator/SVarCControl.mo]: it is intended to have default values for these variables but not exactly that way: fixed

added selectModeAuto0 and setModeManual0 with default values that can be overwritten if needed

**Pull request merged**